stdlib/public/core/ManagedBuffer.swift

```diff
@@ -106,8 +106,10 @@ extension ManagedBuffer {
   /// Call `body` with an `UnsafeMutablePointer` to the stored
   /// `Header`.
   ///
-  /// - Note: This pointer is valid only for the duration of the
-  ///   call to `body`.
+  /// - Note: This pointer is fixed when the buffer is created, but it is only valid
```

Comment:
The header pointer is _not_ guaranteed to be fixed (it might be reimplemented in the future as a temporary whose address is taken, rather than using `Builtin.addressof()` as it does today.) So I don't think I'd assert that here unless the team confirms they're willing to commit to i.t

Reply:
I am assuming that commitment already exists, but if not, I am requesting that commitment now.

Reply:
The pointer is stable right now because `header` is accessed via `Builtin.addressof()` but that is something the team wants to reconsider. Using `&self.header` instead would not produce a stable pointer.

Reply:
However `header` is accessed, I hope the core team agrees that access to `header` must be fast. To me, that means it must not copy. Even if we don’t extend this guarantee to _all_ classes’s properties, I would hope to guarantee it for `ManagedBuffer`.

Reply:
If you want fast access, you can just say `buffer.header`. That's just an indirect load and is as cheap as it could be. If you want to access a mutable pointer to a value in Swift, it needs to be accessed within the constraints of the Swift type system and needs to respect the law of exclusivity. That may involve a copy, or it may be optimized to a simple address-of operation.

The current implementation uses `Builtin.addressof()`, which per @atrick:

> withUnsafe[Mutable]Pointers does not currently enforce exclusive access to the ManagedBuffer's header property. That could actually cause code to miscompile. It only works correctly if the user of the API avoids accessing the header property inside the closure body.

Reply:
Discussed off of GitHub and I believe Andew and Joe are okay with committing to a stable address here.